File: app/src/main/java/zyon/notifier/dialog/EditDialogActivity.java
package zyon.notifier.dialog;

import android.app.Activity;
import android.content.Intent;
import android.database.sqlite.SQLiteDatabase;
import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.widget.Button;
import android.widget.EditText;

import com.chiralcode.colorpicker.ColorPickerDialog;

import zyon.notifier.DBHelper;
import zyon.notifier.R;
import zyon.notifier.service.NotiService;

import static zyon.notifier.MainActivity.TABLE_NAME;

public class EditDialogActivity extends Activity {

    // 데이터베이스
    SQLiteDatabase DB;

    String title_string;
    String text_string;
    String color_string;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        setContentView(R.layout.dialog_edit);

        // 데이터베이스
        DB = new DBHelper(this).getWritableDatabase();

        Intent intent = getIntent();
        String title = intent.getStringExtra("title");
        String text = intent.getStringExtra("text");
        String color = intent.getStringExtra("color");
        final long noti_id = intent.getLongExtra("noti_id", -1);

        color_string = color;
        final View color_add_choose = findViewById(R.id.color_choose_add);
        final View color_add_preview = findViewById(R.id.color_preview_add);
        color_add_preview.setBackgroundColor( Color.parseColor(color_string) );

        final EditText DialogTitle = findViewById( R.id.dialog_title );
        final EditText DialogText = findViewById( R.id.dialog_text );

        DialogTitle.setText( title );
        DialogText.setText( text );

        color_add_choose.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                int initialColor = Color.parseColor(color_string);
                ColorPickerDialog colorPickerDialog = new ColorPickerDialog(EditDialogActivity.this, initialColor, new ColorPickerDialog.OnColorSelectedListener() {
                    @Override
                    public void onColorSelected(int color) {

                        color_string = String.format("#%06X", (0xFFFFFF & color));
                        color_add_preview.setBackgroundColor( Color.parseColor(color_string) );

                    }
                });
                colorPickerDialog.show();

            }
        });

        Button add = findViewById(R.id.button_add);
        add.setOnClickListener(new Button.OnClickListener() {
            @Override
            public void onClick(View view) {

                title_string = DialogTitle.getText().toString();
                text_string = DialogText.getText().toString();

                // 데이터베이스 수정
                DB.execSQL( "DELETE FROM " + TABLE_NAME + " WHERE _id = " + noti_id + ";" );
                DB.execSQL( "INSERT INTO " + TABLE_NAME + " VALUES ( " + noti_id + ", '" + title_string + "', '" + text_string + "', '" + color_string + "' );" );

                // 알림 생성
                startService(
                        new Intent(EditDialogActivity.this, NotiService.class)
                                .putExtra("id", (int)noti_id+"")
                                .putExtra("title", title_string)
                                .putExtra("text", text_string)
                                .putExtra("color", color_string)
                );

                setResult(RESULT_OK, new Intent());
                finish();

            }
        });

        Button cancel = findViewById(R.id.button_cancel);
        cancel.setOnClickListener(new Button.OnClickListener() {
            @Override
            public void onClick(View view) {

                finish();

            }
        });

    }

}

File: app/src/main/java/zyon/notifier/dialog/EditDialogActivity.kt
package zyon.notifier.dialog

import android.app.Activity
import android.content.DialogInterface
import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.view.Window

import androidx.appcompat.app.AlertDialog

import kotlinx.android.synthetic.main.dialog_edit.*

import zyon.notifier.DBHelper
import zyon.notifier.MainActivity
import zyon.notifier.R
import zyon.notifier.service.NotiService

import com.chiralcode.colorpicker.ColorPicker

class EditDialogActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        setContentView(R.layout.dialog_edit)

        // database
        val db = DBHelper(this).writableDatabase
        val intent = intent
        var title = intent.getStringExtra("title")
        var text = intent.getStringExtra("text")
        val color = intent.getStringExtra("color")
        val notiId = intent.getLongExtra("noti_id", -1)
        var colorString = color

        color_preview_add.setBackgroundColor(Color.parseColor(colorString))

        dialog_title.setText(title)
        dialog_text.setText(text)

        color_choose_add.setOnClickListener {

            // open color picker dialog
            val initialColor = Color.parseColor(colorString)

            val alertDialogBuilder = AlertDialog.Builder(this@EditDialogActivity, R.style.DialogTheme)

            val colorPickerView = ColorPicker(this)
            colorPickerView.color = initialColor
            alertDialogBuilder.setView(colorPickerView)

            val onClickListener = DialogInterface.OnClickListener { dialog, which ->
                when (which) {

                    DialogInterface.BUTTON_POSITIVE -> {

                        val selectedColor: Int = colorPickerView.color

                        colorString = String.format("#%06X", 0xFFFFFF and selectedColor)
                        color_preview_add.setBackgroundColor(Color.parseColor(colorString))

                    }

                    DialogInterface.BUTTON_NEGATIVE -> dialog.dismiss()

                }
            }
            alertDialogBuilder.setPositiveButton(this.getString(android.R.string.ok), onClickListener)
            alertDialogBuilder.setNegativeButton(this.getString(android.R.string.cancel), onClickListener)

            val colorPickerDialog = alertDialogBuilder.create()
            colorPickerDialog.show()

        }

        button_add.setOnClickListener {

            title = dialog_title.text.toString()
            text = dialog_text.text.toString()

            // update database
            db.execSQL("DELETE FROM " + MainActivity.TABLE_NAME + " WHERE _id = " + notiId + ";")
            db.execSQL("INSERT INTO " + MainActivity.TABLE_NAME + " VALUES ( " + notiId + ", '" + title + "', '" + text + "', '" + colorString + "' );")

            // create notification
            val notifyIntent = Intent(this@EditDialogActivity, NotiService::class.java)
            notifyIntent.putExtra("id", notiId.toString())
            notifyIntent.putExtra("title", title)
            notifyIntent.putExtra("text", text)
            notifyIntent.putExtra("color", colorString)
            startService(notifyIntent)

            setResult(RESULT_OK, Intent())
            finish()

        }

        button_cancel.setOnClickListener { finish() }

    }

}
